tracer/src/Datadog.Trace/Debugger/Upload/DiagnosticsUploadApi.cs

```diff
@@ -60,21 +61,23 @@ public override async Task<bool> SendBatchAsync(ArraySegment<byte> data)
 
         private Task<IApiResponse> PostAsync(string uri, ArraySegment<byte> data)
         {
-            if (uri.Contains("diagnostics"))
+            var isLegacy = uri.Contains(LegacyEndpoint);
```

Comment:
I'm thinking `Contains` is still a bit unnecessarily lenient here 😅 Is there any reason not to do: 
```csharp
            var isLegacy = uri.Equals(LegacyEndpoint, StringComparison.OrdinalIgnoreCase);
```

or at least

```csharp
            var isLegacy = uri.EndsWith(LegacyEndpoint, StringComparison.OrdinalIgnoreCase);
```

Reply:
Related - do you have any tests that cover this? If not, we probably should?

Reply:
That could be a bit problematic, since the `Endpoint` can be changed before sending request, because of it's subscription to `DiscoveryService` change event. Chances are almost zero, but that could happen 🤔